### PR TITLE
test env: add Tox

### PR DESCRIPTION
**Pull request by TSC21**

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -0,0 +1,248 @@
+#
+# We test yukon using python 3.5 - 3.8.
+#
+# The standard version to develop against is 3.8. We limit mypy checking to 3.8.
+#
+[tox]
+envlist = {py35,py36,py37,py38}-{test,nnvg,doctest},noyaml,lint,mypy,report,docs,package
```
Comment (thirtytwobits) [inline tox.ini:7]: Python 3.6 and newer for Yukon.

Comment (TSC21) [reply tox.ini:7]: Got it!

File: src/yukon/backend/setup.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
+#!/usr/bin/env python3
```
Comment (thirtytwobits) [inline src/yukon/backend/setup.py:1]: The setup.py and setup.cfg should be in the parent folder of `src`

File: src/yukon/backend/setup.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+from typing import Dict
+
+if int(setuptools.__version__.split('.')[0]) < 30:
+    print('A newer version of setuptools is required. The current version does not support declarative config.',
+          file=sys.stderr)
+    sys.exit(1)
+
+version = {}  # type: Dict
+with open('src/api/version.py') as fp:
```
Comment (thirtytwobits) [inline src/yukon/backend/setup.py:19]: version.py isn't at this path.

Comment (thirtytwobits) [reply src/yukon/backend/setup.py:19]: nevermind

File: src/yukon/backend/setup.py (hunk 2 above)
Comment (thirtytwobits) [reply src/yukon/backend/setup.py:1]: sorry. I meant the tox.ini should be next to these.

File: src/yukon/backend/setup.cfg [hunk 4]
```diff
@@ -1,5 +1,5 @@
 [metadata]
-name = yukon/backend
+name = yukon
 version = attr: api.__version__
 author = UAVCAN Development Team
 author_email = zarkopafilis@gmail.com
```
Comment (thirtytwobits) [inline src/yukon/backend/setup.cfg:5]: use uavcan-maintainers@googlegroups.com here

File: src/yukon/backend/test/test_app.py [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+# Copyright (C) 2020  UAVCAN Development Team  <uavcan.org>
+#               2020  dronesolutions.io. All rights reserved.
+# This software is distributed under the terms of the MIT License.
+#
+
+from src.devserv.mock_responses import ServerSentEvent
+import pytest
+
+
+@pytest.mark.parametrize('event', [('test_event')])
```
Comment (thirtytwobits) [inline src/yukon/backend/test/test_app.py:12]: nice

File: tox.ini [hunk 6]
```diff
@@ -0,0 +1,261 @@
...
+    coverage html -d {envtmpdir}
+    coverage xml -o {envtmpdir}/coverage.xml
+
+
+[testenv:mypy]
+basepython = python3.8
+deps = 
+    mypy
+    lxml
+mypy --strict --config-file=setup.cfg api test
```
Comment (thirtytwobits) [inline tox.ini:191]: omit this line (not sure where it came from).

**Pull request merged**